WalletWasabi.Fluent/CrashReport/CrashReporter.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Diagnostics;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Fluent.CrashReport
+{
+	public class CrashReporter
+	{
+		private const int MaxRecursiveCalls = 5;
+		public int Attempts { get; private set; }
+		public string? Base64ExceptionString { get; private set; } = null;
+		public bool HadException { get; private set; }
+		public SerializableException? SerializedException { get; private set; }
+
+		public void TryInvokeIfRequired()
+		{
+			if (!HadException)
+			{
+				return;
+			}
+
+			try
+			{
+				if (Attempts >= MaxRecursiveCalls)
+				{
+					throw new InvalidOperationException($"The crash report has been called {MaxRecursiveCalls} times. Will not continue to avoid recursion errors.");
+				}
+
+				var args = ToCliArguments();
+
+				var path = Process.GetCurrentProcess()?.MainModule?.FileName;
+				if (string.IsNullOrEmpty(path))
+				{
+					throw new InvalidOperationException($"Invalid path: '{path}'");
+				}
+
+				ProcessStartInfo startInfo = ProcessStartInfoFactory.Make(path, args);
+				using Process? p = Process.Start(startInfo);
+			}
+			catch (Exception ex)
+			{
+				Logger.LogWarning($"There was a problem while invoking crash report:{ex.ToUserFriendlyString()}.");
+			}
+		}
+
+		public string ToCliArguments()
+		{
+			if (string.IsNullOrEmpty(Base64ExceptionString))
+			{
+				throw new InvalidOperationException($"The crash report exception message is empty.");
+			}
+
+			return $"crashreport -attempt=\"{Attempts + 1}\" -exception=\"{Base64ExceptionString}\"";
+		}
+
+		public bool TryProcessCliArgs(string[] args)
+		{
+			if (args.Length < 3)
+			{
+				return false;
+			}
+
+			if (args[0].Contains("crashreport") && args[1].Contains("-attempt=") && args[2].Contains("-exception="))
```

Comment:
```csharp
[
  "crashreport",
  "-invalidParameter \"-attempt=fail\"",
  "-invalidParameter2 \"-exception=fail\"",
]
```
?

:)

Reply:
hmm, what is this?

Reply:
It's an example of `args` that will presumably lead to crash.

Reply:
Crashreport args are generated by Wasabi and processed by Wasabi. The test cases are covering validating the workflow. If somebody manually tries to hack it then Wasabi will crash. 
Isn't that good?

Reply:
IDK I would use that Mono thing to parse those arguments correctly.